### PR TITLE
[TEST] 전체 단위 테스트 코드 구현 및 회귀 테스트 기반 마련

### PR DESCRIPTION
**Pull request by qldrh112**
## 📝 변경 사항 요약
-  tests/unit/ 디렉터리 내 모든 핵심 모듈(pdf_loader, preprocessor, ocr_engine, postprocessor, formatters, validators, excel_builder 등)에 대한 단위 테스트 작성을 완료했습니다.
- 의존성을 분리한 가벼운 단위 테스트 환경을 구축하여, 향후 코드 변경 시 즉각적이고 안정적인 회귀 테스트가 가능하도록 기반을 마련했습니다.
- 요구사항 및 기능 명세에 맞춰 [Tests.csv 문서](https://docs.google.com/spreadsheets/d/1qUclaf1vGhyrnFFPQoa0_MShbkzTYcgHezVEV9st0Xw/edit?gid=95924164#gid=95924164&range=A1)를 업데이트했습니다. 데이터 정제 과정의 추적성을 확보하기 위해 TEST-006(금융상품 포매터), TEST-007(대출거래 포매터), TEST-008(유효성 검사기) 항목을 대상 기능인 FUNC-005(OCR 결과 후처리)에 논리적으로 매핑 및 연결했습니다. 

<br>

## 🧐 리뷰어가 중점적으로 봐야 할 부분
- FUNC-005 후처리 과정에 포함된 formatters.py와 validators.py의 다양한 엣지 케이스(Edge case)들이 테스트 코드에 충분히 반영되었는지 중점적으로 리뷰해 주시면 감사하겠습니다.
- 전체 모의 객체(Mock) 적용이 적절하게 이루어졌는지 확인 부탁드립니다.

<br>

## ✅ 이 변경 사항을 어떻게 테스트했나요?
로컬 환경에서 다음 명령어를 통해 새롭게 작성된 단위 테스트 모음이 모두 통과하는지 확인할 수 있습니다.
```
# 단위 테스트만 빠르게 실행하여 회귀 검증
pytest tests/unit

# 기존에 작성된 통합 테스트(integration)를 포함한 전체 파이프라인 검증
pytest
```

<br>

## 📸 스크린샷 (선택 사항)

image  

- **test_full_pipeline_accuracy**: 추가 확인 필요
- **TestValidatorsUnit**: 아직 구현되지 않았으나  Tests.csv에 정의한 내용을 기반으로 초안 생성
- **test_ocr_pipeline_integration**: 통합 테스트 항목이므로 integration 테스트로 이동 필요 https://github.com/dongtan-91-dong-welfare-center/audit_inquiry_automation/issues/29

<br>

---

## 🧑‍💻 PR 작성자 셀프 체크리스트

- [ ] Issue 번호를 연결하고 PR 제목에 핵심 내용을 요약했습니다.
- [x] `CONTRIBUTING.md`의 Git 워크플로우를 따랐습니다.
- [x] `.editorconfig`에 정의된 코딩 스타일을 준수했습니다.
- [x] 제 코드를 스스로 다시 한번 리뷰했습니다.
- [ ] (필요하다면) 관련 문서(예: README)를 업데이트했습니다.
- [x] (필요하다면) 기능에 대한 단위 테스트 코드를 추가했습니다.

### CODE REVIEW

File: tests/unit/test_excel_builder.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    @pytest.mark.unit
+    @patch('os.path.exists')
+    @patch('shutil.copy')
+    @patch('openpyxl.load_workbook')
+    def test_export_data_append_logic(self, mock_load_wb, mock_copy, mock_exists):
+        """데이터 삽입 및 조회처 꼬리표 부착 로직 검증"""
+
+        # 1. 파일 시스템 상태 모의: os.path.exists
+        # 처음 호출(출력 파일 확인) 시 False 반환 -> 두 번째 호출(템플릿 파일 확인) 시 True 반환
+        # if not os.path.exists(output_path) -> False
```
Comment (PDW-accountant) [inline tests/unit/test_excel_builder.py]: not을 제거하는게 논리적으로 옳습니다.

File: tests/conftest.py [hunk 2]
```diff
@@ -4,21 +4,51 @@
...
+import numpy as np
 import pytest
 from pathlib import Path
+from unittest.mock import MagicMock
 
 # 프로젝트 루트 디렉토리를 sys.path에 추가
 # sys.path: 파이썬이 모듈이나 패키지를 찾을 때 훑어보는 경로 목록
-sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), '../')))
-
+# sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), '../')))
```
Comment (PDW-accountant) [inline tests/conftest.py:15]: pyproject.toml에서 경로를 설정하고 있으면 해당 주석은 삭제해도 괜찮지 않나요? 이해 목적으로 남기신건가요?

File: tests/conftest.py [hunk 3]
```diff
@@ -37,10 +67,10 @@ def scan_pdf_path():
...
     return str(path)
 
-# Pytest 마커 정의 (커스텀 태그 등록)
-def pytest_configure(config):
-    """
-    커스텀 마커를 등록하여 'pytest -m integration' 처럼 실행할 수 있게 합니다.
-    """
-    config.addinivalue_line("markers", "integration: 실제 PDF 파일을 사용하는 무거운 통합 테스트")
-    config.addinivalue_line("markers", "unit: Mock을 사용하는 가벼운 단위 테스트")
+# # Pytest 마커 정의 (커스텀 태그 등록)
```
Comment (PDW-accountant) [inline tests/conftest.py:73]: 동일한 사항입니다.

File: tests/unit/test_excel_builder.py (hunk 1 above)
Comment (gyuseok57kg) [reply tests/unit/test_excel_builder.py]: > not을 제거하는게 논리적으로 옳습니다.

@PDW-accountant 님께서 구현하신대로 출력 결과가 없으면 템플릿의 유무를 확인하여 처리하는 로직이 맞습니다.

그러나, 단위 테스트 단계에서는 실제 데이터를 활용하지 않습니다.
따라서, 실제로 조건문을 실행해서 분기하는 것이 아니라 mock_exists.side_effect에 미리 조건문 결과를 담아두는 것입니다.

예를 들어 side_effect = [False, True]로 설정하여 첫 번째 호출인 출력 파일 확인은 False, 두 번째 호출인 템플릿 파일 확인은 True를 반환하도록 조작하는 원리입니다.

그래서 결국 이 테스트는 출력 파일이 없고 템플릿이 있을 때 정상적으로 엑셀을 만들어서 반환하는지 확인하는 함수입니다.

File: tests/conftest.py (hunk 2 above)
Comment (gyuseok57kg) [reply tests/conftest.py:15]: > pyproject.toml에서 경로를 설정하고 있으면 해당 주석은 삭제해도 괜찮지 않나요? 이해 목적으로 남기신건가요?

말씀하신 바와 같이 이전에는 line 14 코드를 사용했었다는 이해 목적으로 남겼습니다. 다만, 맥락이 부족하여 주석을 추가하겠습니다.

File: tests/conftest.py (hunk 3 above)
Comment (gyuseok57kg) [reply tests/conftest.py:73]: > 동일한 사항입니다.

해당 내용은 삭제하고, pyproject.toml에 기재하겠습니다.

File: tests/unit/test_excel_builder.py (hunk 1 above)
Comment (PDW-accountant) [reply tests/unit/test_excel_builder.py]: 해당 if문은 "output_path"가 없어서 false를 반환해야 하므로 if os.path.exists(output_path) -> False가 되어야 하는거 아닌가요? not이 있으면 반환값이 true가 되어야 할 것 같은데요.

Comment (gyuseok57kg) [reply tests/unit/test_excel_builder.py]: `os.path.exists()` 자체는 False를 반환해야 `if not` 조건을 통해 True가 되어 블록 내부로 진입해야  의도한 대로 출력 파일이 상황을 가정하고 테스트할 수 있습니다.

따라서, 로직 자체는 문제는 없으나 주석이 혼동을 유발하는 상황입니다.

따라서 주석을 아래와 같이 변경하여 모호함을 줄였습니다.
```
# os.path.exists(output_path) -> False를 입력하여 조건문의 평가를 True로 진행함
# os.path.exists(template_path) -> True를 입력하여 조건문의 평가를 True로 진행하여 템플릿 복사가 일어나게 함
```

**Review by PDW-accountant — APPROVED** [review]
리뷰사항을 반영하여 변경된 부분들을 모두 확인하였습니다.
